### PR TITLE
THEMES-707: Lead Art Block v2

### PR DESCRIPTION
**Pull request by vgalatro**
## Description

This PR converts the lead art block to the new v2 architecture.

## Jira Ticket

- [THEMES-707](https://arcpublishing.atlassian.net/browse/THEMES-707)

## Acceptance Criteria

- Lead Art block is migrated to the new architecture.
  - Lead Art can be a Video, Image or Gallery; those blocks should be converted first

## Test Steps

Feature Pack PR -

Use Themes Internal Sandbox for content base

1. Checkout this branch `git checkout THEMES-707-lead-art-v2`
2. Checkout the feature pack branch `git checkout THEMES-707-lead-art-v2`
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/lead-art-block`
4. Edit the article right rail template to use the new lead art block.
5. Use the default block settings, and test with stories that have the different lead art types (image, gallery, video, and raw HTML).
6. Test that the 'Hide...' options have the desired effect.
7. Test that the 'Playthrough' and 'Autoplay' options effect the video lead art type.
8. Test that the image loading strategy is passed through to the `Image` component for the image lead art type.

## Dependencies or Side Effects

- Feature pack PR: https://github.com/WPMedia/arc-themes-feature-pack/pull/365

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/lead-art-block/features/leadart/default.jsx [hunk 1]
```diff
@@ -1,226 +1,257 @@
...
-				);
+		if (lead_art.type === "image") {
+			return (
+				<MediaItem
+					caption={!hideCaption ? lead_art.caption : null}
+					credit={!hideCredits ? formatCredits(lead_art.credits) : null}
+					title={!hideTitle ? lead_art.subtitle : null}
+				>
+					<Button
+						iconLeft={<Icon name="Fullscreen" width="100%" height="100%" fill="#6B6B6B" />}
```
Comment (matthewroach) [inline blocks/lead-art-block/features/leadart/default.jsx]: Width and height props are generating errors - you could get away without them here as the Icon size is set in CSS

File: blocks/lead-art-block/features/leadart/default.jsx [hunk 2]
```diff
@@ -1,226 +1,257 @@
...
+						label={`${
```
Comment (matthewroach) [inline blocks/lead-art-block/features/leadart/default.jsx]: All text should be using translations strings

File: blocks/lead-art-block/features/leadart/default.jsx [hunk 3]
```diff
@@ -1,226 +1,257 @@
...
-								lightBoxHeight={0}
-								breakpoints={getProperties(arcSite)?.breakpoints}
-								resizerURL={getProperties(arcSite)?.resizerURL}
-								resizedImageOptions={lead_art.resized_params}
-								loading={allowedImageLoadingStrategy} // eager by default, otherwise lazy
+					}
+					adElement={/* istanbul ignore next */ <AdBlock />}
+					adInterstitialClicks={interstitialClicks}
+					nextButton={
+						<Carousel.Button id={`${BLOCK_CLASS_NAME}-${lead_art._id}`} label="Next Slide">
```
Comment (matthewroach) [inline blocks/lead-art-block/features/leadart/default.jsx]: Text string should use translations

File: blocks/lead-art-block/features/leadart/default.jsx [hunk 4]
```diff
@@ -1,226 +1,257 @@
...
-						displayCredits={!hideCredits}
-					/>
-				);
-			}
-
-			return null;
+						</Carousel.Button>
+					}
+					previousButton={
+						<Carousel.Button id={`${BLOCK_CLASS_NAME}-${lead_art._id}`} label="Previous Slide">
```
Comment (matthewroach) [inline blocks/lead-art-block/features/leadart/default.jsx]: Text string should use translations

File: .storybook/themes/news.scss [hunk 5]
```diff
@@ -1594,6 +1594,86 @@
 						),
 					),
 				),
+				lead-art: (
+					margin-bottom: var(--global-spacing-5),
+					components: (
+						button: (
+							border: none,
```
Comment (matthewroach) [inline .storybook/themes/news.scss]: This is causing regression on the header nav block

**Review by matthewroach — CHANGES_REQUESTED** [review]

**Comment by vgalatro** [comment]
@matthewroach , I've addressed the feedback for this PR, ready for review again.

File: blocks/lead-art-block/features/leadart/default.jsx [hunk 6]
```diff
@@ -1,226 +1,265 @@
...
+					<Button
+						iconLeft={<Icon name="Fullscreen" fill="#6B6B6B" />}
+						label={
+							!isOpen
+								? phrases.t("lead-art-block.fullscreen-enter")
+								: phrases.t("lead-art-block.fullscreen-exit")
+						}
+						onClick={toggleFullScreen}
+					>
+						<span>{buttonLabel}</span>
```
Comment (matthewroach) [inline blocks/lead-art-block/features/leadart/default.jsx]: No need for `<span>` here - The button uses a `<span>` which ends up giving us nested elements

```suggestion
						{buttonLabel}
```

Comment (vgalatro) [reply blocks/lead-art-block/features/leadart/default.jsx]: Got it, removed the span.

**Review by matthewroach — APPROVED** [review]
Nice work, this is now ready for design QA @haganka

**Pull request merged**